src/deepsparse/server/system_logging.py

```diff
@@ -12,24 +12,47 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from os import getpid
 from typing import Any
 
+import psutil
 from deepsparse import Pipeline
+from deepsparse.loggers import BaseLogger, MetricCategories
 
 
 __all__ = ["log_resource_utilization", "log_request_details"]
 
 
-def log_resource_utilization(pipeline: Pipeline, **kwargs: Any):
+def log_resource_utilization(server_logger: BaseLogger):
     """
-    Scope for 1.4:
-    - CPU utilization overall
-    - Memory available overall
-    - Memory used overall (shall we continuously log this?
-      this will be a constant value in time)
-    - Number of core used by the pipeline
+    Logs the resource utilization of the server process.
+    This includes:
+    - CPU utilization
+    - Memory utilization
+    - Total memory available
+
+    :param server_logger: the logger to log the metrics to
     """
-    pass
+    process = psutil.Process(getpid())
```

Comment:
is this already in our setup.py dependencies?

Reply:
haha, no. good catch.